core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -729,16 +743,16 @@ public static Builder builder(MetaData metaData) {
     }
 
     /** All known byte-sized cluster settings. */
-    public static final Set<String> CLUSTER_BYTES_SIZE_SETTINGS = ImmutableSet.of(
+    public static final Set<String> CLUSTER_BYTES_SIZE_SETTINGS = unmodifiableSet(newHashSet(
```

Comment:
This can be left as a TODO for another PR but I don't see using Sets.newHashSet as an improvement, when we can directly use the JDK APIs and do `new HashSet<>(Arrays.asList(...))`